### ISSUE TITLE
HTTPS question

### ISSUE THREAD

**Issue by bonky42** [F5Networks/k8s-bigip-ctlr#789]
#### Description
Hello, I've got a weird comportment with https, just imagine this following yaml:
```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: f5-dev-it-transverse
  namespace: dev-it-transverse
  annotations:
    kubernetes.io/ingress.class: "f5"
    virtual-server.f5.com/ip: "*******"
    virtual-server.f5.com/partition: "******"
    virtual-server.f5.com/health: |
      [
        {
          "path":     "foo.foo.com/",
          "send":     "GET /?monitorf5 HTTP/1.1\r\n",
          "interval": 5,
          "timeout":  10
        }
      ]
    ingress.kubernetes.io/ssl-redirect: "true"
    ingress.kubernetes.io/allow-http: "false"
spec:
  tls:
  - secretName: ************************
  rules:
  - host: foo.foo.com
    http:
      paths:
      - backend:
          serviceName: ******
          servicePort: ******
        path: /

````
if I go to _http://foo.foo.com_ I will be redirected to _https://foo.foo.com_ the same with a / at the end
but if I go to _http://foo.foo.com/toto_ I will stay in http.

If I look to the irule it make sense
Where am I missing something? Isn't it suppose to redirect all request to https which ever the path is?

For a workaround I tried to put /* for the path in the yaml, it works, but in this case I got this error :
`2018/11/06 13:21:28 [ERROR] [2018-11-06 13:21:28,642 f5_cccl.resource.resource ERROR] HTTP error(400): CCCL resource(ApiPolicy) /xxxx/ingress_xxxxxxxxxxxxxx_80.
2018/11/06 13:21:28 [ERROR] [2018-11-06 13:21:28,642 f5_cccl.service.manager ERROR] F5CcclResourceRequestError - 400 Unexpected Error: Bad Request for uri: https://xxxxxxxxxxxxxx:443/mgmt/tm
/ltm/policy/
2018/11/06 13:21:28 [INFO] Text: u'{"code":400,"message":"01071748:3: Policy \'/xxxx/ingress_xxxxxxxxxxxxxx\', rule \'ingress_xxxxxxxxxxxxxx_*_ingress_xxxxxxxxxxxxxx\'; invalid rule name, value \'ingress_xxxxxxxxxxx_*_ingress_xxxxxxxxxxxxxxx\'.","errorStack":[],"apiError":3}'
2018/11/06 13:21:28 [ERROR] [2018-11-06 13:21:28,642 f5_cccl.service.manager ERROR] Resource /xxxx/ingress_xxxxxxxxxxxxxx_80 creation error, requeuing task...`

With the name of the policy it's normal, can you help me on this matter?
#### Kubernetes Version
1.9

#### Controller Version
1.6.0

#### BIG-IP Version
Product     BIG-IP
  Version     12.1.3
  Build       0.0.378
  Edition     Final

#### Diagnostic Information
N/A

**Comment by bonky42** [comment]
Hi @veeramalla-f5 ,

If I follow you, if I want to share the same IP for multiple web sites ex (toto.my-company.com, titi.my-company.com), I have to put each and every possible path to have all of them under https. On the contrary, if I use one ingress by web site to have all covered by https, I can't use the same IP for all of them.
Am I correct?
If yes, that's a huge point against this controller, in a world where everything is going https, I can't understand why when we ask the path / to be in https, all the subsidiaries or not. I can't follow all the path the developers will create for their needs.

**Comment by veeramalla-f5** [comment]
Hi @bonky42 , 

 My bad. I have misunderstood the issue. I have figured out a solution for this fix and it will be available very soon.

Regards,
Abhishek Veeramalla.

**Comment by bonky42** [comment]
no pb, thanks for the quick answer, I'm waiting for the fix :)

**Comment by AbhishekPy** [comment]
Fix is available as part of 1.7.1
